File: April/Abiji-2020/Golang /Find If Path Exits in Graph/find if path exits in graph.go
func dfs(vertex int, graph [][]int, vis []int) {
	vis[vertex] = 1
	for _, child := range graph[vertex] {
		if vis[child] == 0 {
			dfs(child, graph, vis)
		}
	}
}

func validPath(n int, edges [][]int, source, destination int) bool {
	graph := make([][]int, n)
	vis := make([]int, n)

	for _, edge := range edges {
		graph[edge[0]] = append(graph[edge[0]], edge[1])
		graph[edge[1]] = append(graph[edge[1]], edge[0])
	}

	dfs(source, graph, vis)

	return vis[destination] == 1
}

File: April/Abiji-2020/C++/Find If Path Exists in Graph/find if path exits in graph.cpp
class Solution {
public:
    // Recursive DFS function to explore the graph
    void dfs(int vertex, vector<vector<int>>& graph, vector<int>& vis) {
        vis[vertex] = 1; // Mark the current vertex as visited
        // Iterate over all adjacent vertices (children)
        for (auto child : graph[vertex]) {
            if (!vis[child]) { // If the child vertex has not been visited
                dfs(child, graph, vis); // Recursively visit the child
            }
        }
    }

    // Function to check if there's a valid path between source and destination
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        vector<vector<int>> graph(n); // Adjacency list for the graph
        vector<int> vis(n, 0); // Visited array to keep track of visited vertices, initialized to 0
        
        // Building the graph from the edge list
        for (int i = 0; i < edges.size(); i++) {
            graph[edges[i][0]].push_back(edges[i][1]); // Add an edge from vertex edges[i][0] to edges[i][1]
            graph[edges[i][1]].push_back(edges[i][0]); // Since the graph is undirected, add the reverse edge as well
        }

        // Start DFS from the source vertex
        dfs(source, graph, vis);

        // Check if the destination vertex has been visited
        return vis[destination] == 1;
    }
};
